starboard/nplb/posix_compliance/posix_clock_nanosleep_test.cc

```diff
@@ -0,0 +1,455 @@
+// Copyright 2025 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <errno.h>
+#include <signal.h>
+#include <sys/time.h>  // For gettimeofday()
+#include <time.h>
+#include <unistd.h>  // For syscall
+
+#include "build/build_config.h"
+#include "starboard/configuration_constants.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace starboard {
+namespace nplb {
+namespace {
+
+// Small duration for sleeps that should be short or return immediately.
+const long kShortSleepNs = 1'000'000;  // 1 millisecond
+const long kShortSleepUs = kShortSleepNs / 1'000;
+// A slightly longer duration for testing actual sleep.
+const long kTestSleepNs = 50'000'000;  // 50 milliseconds
+const long kTestSleepUs = kTestSleepNs / 1'000;
+// A very long duration for testing interruption.
+const long kLongSleepSec = 10;  // 10 seconds
+// Duration for alarm to trigger, should be less than kLongSleepSec
+const unsigned int kAlarmSec = 1;
+
+// Signal handler for EINTR tests (can be used for SIGALRM)
+void InterruptSignalHandler(int signum) {
+  // Do nothing, just here to interrupt sleep
+  (void)signum;  // Suppress unused parameter warning
+}
+
+class PosixClockNanosleepTest : public ::testing::Test {
+ protected:
+  struct sigaction old_sa_sigalrm;  // To store original SIGALRM handler
+
+  void SetUp() override {
+    // Ensure no stale errno
+    errno = 0;
+    // Clear any pending alarm from a previous failed test or other source
+    alarm(0);
+
+    // Store current SIGALRM handler
+    struct sigaction current_sa;
```

Comment:
Does this somehow get implicitly set in the following call? Right now it looks unused